src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -643,6 +628,28 @@ REDHAWK_PALEXPORT bool REDHAWK_PALAPI PalStartBackgroundWork(_In_ BackgroundCall
     int st2 = pthread_attr_destroy(&attrs);
     ASSERT(st2 == 0);
 
+    if (highPriority && st == 0)
+    {
+        int policy = SCHED_RR;
+        int maxPriority = sched_get_priority_max(policy);
```

Comment:
CoreCLR computes the policy using `pthread_getschedparam`. Is it safe to hard-code this to `SCHED_RR` ?

https://github.com/dotnet/runtime/blob/6991ccd1e8cdfb640a74be7db54e81f456406d1f/src/coreclr/src/pal/src/thread/thread.cpp#L1143

Reply:
As the thread was just created the policy would be SCHED_OTHER. SCHED_OTHER does only allow 0 as priority. To get the thread to a higher priority than all the other threads we need to use one of the scheduler that has a priority range. This limits us to SCHED_FIFO or SCHED_RR. I used SCHED_RR because it has a time quantum.

Reply:
Wouldn't the initial policy rather match the policy of the process? So e.g. when you launch the process using the chrt tool with the `-f` or `-r` options, it seems the thread would start with SCHED_FIFO or SCHED_RR.

Reply:
Thanks, I wasn't aware of this. In this case I really need to check the current policy first and only migrate to another one if the current one doesn't support different priorities